Narrowing conversion warnings by the compiler

**Describe the bug**
Inspired by the comment in https://github.com/LDMX-Software/ldmx-sw/pull/1198#issuecomment-1717288314 when I compile ldmx-sw I see the following narrowing conversions in ldmx-sw:

In `HcalGeometry.cxx`
```
/Users/tav/Documents/1Research/LDMX/ldmx-sw/DetDescr/src/DetDescr/HcalGeometry.cxx: In member function 'void ldmx::HcalGeometry::buildStripPositionMap()':
/Users/tav/Documents/1Research/LDMX/ldmx-sw/DetDescr/src/DetDescr/HcalGeometry.cxx:156:31: warning: narrowing conversion of 'section' from 'int' to 'unsigned int' [-Wnarrowing]
  156 |         const ldmx::HcalID id{section, layer, strip};
      |                               ^~~~~~~
/Users/tav/Documents/1Research/LDMX/ldmx-sw/DetDescr/src/DetDescr/HcalGeometry.cxx:156:40: warning: narrowing conversion of 'layer' from 'int' to 'unsigned int' [-Wnarrowing]
  156 |         const ldmx::HcalID id{section, layer, strip};
      |                                        ^~~~~
/Users/tav/Documents/1Research/LDMX/ldmx-sw/DetDescr/src/DetDescr/HcalGeometry.cxx:156:47: warning: narrowing conversion of 'strip' from 'int' to 'unsigned int' [-Wnarrowing]
  156 |         const ldmx::HcalID id{section, layer, strip};
      |                                               ^~~~~
/Users/tav/Documents/1Research/LDMX/ldmx-sw/DetDescr/src/DetDescr/HcalGeometry.cxx:211:57: warning: narrowing conversion of 'id.ldmx::HcalID::section()' from 'int' to 'unsigned int' [-Wnarrowing]
  211 |                       (getScintillatorLength({id.section(), 2, id.strip()}) -
      |                                               ~~~~~~~~~~^~
/Users/tav/Documents/1Research/LDMX/ldmx-sw/DetDescr/src/DetDescr/HcalGeometry.cxx:211:72: warning: narrowing conversion of 'id.ldmx::HcalID::strip()' from 'int' to 'unsigned int' [-Wnarrowing]
  211 |                       (getScintillatorLength({id.section(), 2, id.strip()}) -
      |                                                                ~~~~~~~~^~
/Users/tav/Documents/1Research/LDMX/ldmx-sw/DetDescr/src/DetDescr/HcalGeometry.cxx:238:57: warning: narrowing conversion of 'id.ldmx::HcalID::section()' from 'int' to 'unsigned int' [-Wnarrowing]
  238 |                       (getScintillatorLength({id.section(), 2, id.strip()}) -
      |                                               ~~~~~~~~~~^~
/Users/tav/Documents/1Research/LDMX/ldmx-sw/DetDescr/src/DetDescr/HcalGeometry.cxx:238:72: warning: narrowing conversion of 'id.ldmx::HcalID::strip()' from 'int' to 'unsigned int' [-Wnarrowing]
  238 |                       (getScintillatorLength({id.section(), 2, id.strip()}) -
```

and in `HgcrocEmulator.cxx`

```
/Users/tav/Documents/1Research/LDMX/ldmx-sw/Tools/src/Tools/HgcrocEmulator.cxx: In member function 'std::vector<ldmx::HgcrocDigiCollection::Sample> ldmx::HgcrocEmulator::noiseDigi(const int&, const double&) const':
/Users/tav/Documents/1Research/LDMX/ldmx-sw/Tools/src/Tools/HgcrocEmulator.cxx:235:17: warning: narrowing conversion of 'pedestal' from 'double' to 'int' [-Wnarrowing]
  235 |     int adc_tm1{pedestal};
      |                 ^~~~~~~~
/Users/tav/Documents/1Research/LDMX/ldmx-sw/Tools/src/Tools/HgcrocEmulator.cxx:240:24: warning: narrowing conversion of '(pedestal + (((const ldmx::HgcrocEmulator*)this)->ldmx::HgcrocEmulator::noise((* & channel)) / gain)' from 'double' to 'int' [-Wnarrowing]
  240 |     int adc_t{pedestal + noise(channel)/gain};
      |               ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~
```

I believe this second one is responsible for the msg @bryngemark referred too.

**To Reproduce**

Just compiling with the latest gcc version

I found one more of this type
```
/sdf/home/t/tamasvami/BugFixes/ldmx-sw/TrigScint/src/TrigScint/QIEDecoder.cxx:182:56: warning: narrowing conversion of '(int)(((unsigned char)((int)(flags >> ((int)trigscint::QIEStream::CID_SKIP_POS)))) & ((unsigned char)((int)mask8<1>::m)))' from 'int' to 'bool' [-Wnarrowing]
  182 |   bool isCIDskipped{(flags >> QIEStream::CID_SKIP_POS) &
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  183 |                     mask8<QIEStream::FLAG_SIZE_BITS>::m};
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
/sdf/home/t/tamasvami/BugFixes/ldmx-sw/TrigScint/src/TrigScint/QIEDecoder.cxx:184:57: warning: narrowing conversion of '(int)(((unsigned char)((int)(flags >> ((int)trigscint::QIEStream::CID_UNSYNC_POS)))) & ((unsigned char)((int)mask8<1>::m)))' from 'int' to 'bool' [-Wnarrowing]
  184 |   bool isCIDunsync{(flags >> QIEStream::CID_UNSYNC_POS) &
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  185 |                    mask8<QIEStream::FLAG_SIZE_BITS>::m};
      |                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~   
/sdf/home/t/tamasvami/BugFixes/ldmx-sw/TrigScint/src/TrigScint/QIEDecoder.cxx:186:59: warning: narrowing conversion of '(int)(((unsigned char)((int)(flags >> ((int)trigscint::QIEStream::CRC1_ERR_POS)))) & ((unsigned char)((int)mask8<1>::m)))' from 'int' to 'bool' [-Wnarrowing]
  186 |   bool isCRC1malformed{(flags >> QIEStream::CRC1_ERR_POS) &
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  187 |                        mask8<QIEStream::FLAG_SIZE_BITS>::m};
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
/sdf/home/t/tamasvami/BugFixes/ldmx-sw/TrigScint/src/TrigScint/QIEDecoder.cxx:188:59: warning: narrowing conversion of '((((int)flags) >> ((int)trigscint::QIEStream::CRC0_ERR_POS)) & ((int)mask8<1>::m))' from 'int' to 'bool' [-Wnarrowing]
  188 |   bool isCRC0malformed{(flags >> QIEStream::CRC0_ERR_POS) &
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  189 |                        mask8<QIEStream::FLAG_SIZE_BITS>::m};
      |                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ 
```

Not sure what to do with it tho, @bryngemark any suggestions on how this was meant to be? Should we just `static_cast` it?

--- a/TrigScint/src/TrigScint/QIEDecoder.cxx
+++ b/TrigScint/src/TrigScint/QIEDecoder.cxx
@@ -63,15 +63,6 @@ void QIEDecoder::produce(framework::Event &event) {
   ldmx_log(debug) << "QIEDecoder: produce() starts! Event number: "
                   << event.getEventHeader().getEventNumber();
 
-  //	ldmx::EventHeader *eh = (ldmx::EventHeader*)event.getEventHeaderPtr();
-  // eh->setIsRealData(isRealData_); //doesn't help
-
-  /*   // comment for now, requires pushing change ot EventHeader
-  event.getEventHeader().setIsRealData(isRealData_);
-  ldmx_log(debug) << "Decoder bool isRealData = " << isRealData_ << " and after
-  attempt of setting it, event header returns " <<
-  event.getEventHeader().isRealData();
-  */
   // turns out this need to be configurable for now, to read real data
   int nSamp = nSamples_;  // QIEStream::NUM_SAMPLES ;
   ldmx_log(debug) << "num samples = " << nSamp;
@@ -127,18 +118,15 @@ void QIEDecoder::produce(framework::Event &event) {
 
   int sigBitsSkip = 6;  // the first 6 bits are part of something else.
   int divisor = TMath::Power(2, 32 - sigBitsSkip);
-  timeSpill =
-      timeSpill % divisor;  // remove them by taking remainder in division by
-                            // the values of the last skipped bit
-  // timeSpill=timeSpill/spillTimeConv_;
+  // remove them by taking remainder in division by
+  // the values of the last skipped bit
+  timeSpill = timeSpill % divisor;
   ldmx_log(debug) << "After taking it mod 2^" << 32 - sigBitsSkip
                   << " (which is " << divisor << ", spill time is "
                   << timeSpill;
   event.getEventHeader().setIntParameter("timeSinceSpill", timeSpill);
-  //  event.getEventHeader().setTimeSinceSpill(timeSpill); //not working
 
   TTimeStamp *timeStamp = new TTimeStamp(timeEpoch);
-  //  timeStamp->SetNanoSec(timeClock); //not sure about this one...
   event.getEventHeader().setTimestamp(*timeStamp);
 
   // trigger ID event number
@@ -155,9 +143,7 @@ void QIEDecoder::produce(framework::Event &event) {
   }
 
   //  ldmx_log(debug) << " got triggerID " << std::bitset<16>(triggerID) ;
-  //  //eventStream.at(0);
-  ldmx_log(debug) << " got triggerID "
-                  << std::bitset<32>(triggerID);  // eventStream.at(0);
+  ldmx_log(debug) << " got triggerID " << std::bitset<32>(triggerID);
 
   if (triggerID != event.getEventHeader().getEventNumber()) {
     // this probably only applies to digi emulation,
@@ -179,17 +165,18 @@ void QIEDecoder::produce(framework::Event &event) {
   */
   uint8_t flags = eventStream.at(QIEStream::ERROR_POS);
 
-  bool isCIDskipped{(flags >> QIEStream::CID_SKIP_POS) &
-                    mask8<QIEStream::FLAG_SIZE_BITS>::m};
-  bool isCIDunsync{(flags >> QIEStream::CID_UNSYNC_POS) &
-                   mask8<QIEStream::FLAG_SIZE_BITS>::m};
-  bool isCRC1malformed{(flags >> QIEStream::CRC1_ERR_POS) &
-                       mask8<QIEStream::FLAG_SIZE_BITS>::m};
-  bool isCRC0malformed{(flags >> QIEStream::CRC0_ERR_POS) &
-                       mask8<QIEStream::FLAG_SIZE_BITS>::m};
+  bool isCIDskipped{static_cast<bool>((flags >> QIEStream::CID_SKIP_POS) &
+                                      mask8<QIEStream::FLAG_SIZE_BITS>::m)};
+  bool isCIDunsync{static_cast<bool>((flags >> QIEStream::CID_UNSYNC_POS) &
+                                     mask8<QIEStream::FLAG_SIZE_BITS>::m)};
+  bool isCRC1malformed{static_cast<bool>((flags >> QIEStream::CRC1_ERR_POS) &
+                                         mask8<QIEStream::FLAG_SIZE_BITS>::m)};
+  bool isCRC0malformed{static_cast<bool>((flags >> QIEStream::CRC0_ERR_POS) &
+                                         mask8<QIEStream::FLAG_SIZE_BITS>::m)};
   // checksum
-  uint8_t referenceChecksum = 0;  // really, this is just empty for now. TODO>
-                                  // implement a checksum set/get
+  // really, this is just empty for now.
+  // TODO: implement a checksum set/get
+  uint8_t referenceChecksum = 0;
   int checksum{(flags >> QIEStream::CHECKSUM_POS) &
                mask8<QIEStream::CHECKSUM_SIZE_BITS>::
                    m};  // eventStream.at(QIEStream::CRC0_ERR_POS)

